models/status.go

```
@@ -219,7 +219,10 @@ func newCommitStatus(sess *xorm.Session, opts NewCommitStatusOptions) error {
 	}
 	has, err := sess.Desc("index").Limit(1).Get(lastCommitStatus)
 	if err != nil {
-		sess.Rollback()
+		err = sess.Rollback()
```

The first error is lost.